File: docs/platform/evidence/phase1-verdicts-2026-09-25/tools/lib/paths.mjs
/*
 * Where the Phase 1 verdict tools read and write. Each path comes from the environment and
 * falls back to a default:
 *
 *   SITE_ROOT  the site repository: the root this folder sits in (tools/../../../../..)
 *   WORK       scratch output, outside the repository: <os tmpdir>/phase1-verdicts-2026-09-25
 *
 * The Python tools read the same variables through lib/paths.py, which adds SWISS_EPHE and
 * JPL_KERNEL. Swiss Ephemeris output is written only under WORK; loading this module fails if
 * WORK is the repository or lies inside it.
 */
import { mkdirSync } from 'node:fs';
import { tmpdir } from 'node:os';
import { dirname, isAbsolute, join, relative, resolve, sep } from 'node:path';
import { fileURLToPath } from 'node:url';

export const TOOLS = resolve(dirname(fileURLToPath(import.meta.url)), '..');
export const SITE_ROOT = resolve(process.env.SITE_ROOT || resolve(TOOLS, '../../../../..'));
export const WORK = resolve(process.env.WORK || join(tmpdir(), 'phase1-verdicts-2026-09-25'));

const fromRoot = relative(SITE_ROOT, WORK);
if (!(fromRoot === '..' || fromRoot.startsWith(`..${sep}`) || isAbsolute(fromRoot))) {
  throw new Error(`WORK (${WORK}) is inside the repository (${SITE_ROOT}); Swiss output must stay outside it`);
}

/** The installed @zodiacs/engine, which run-all.sh checks against vendor/zodiacs-engine-0.1.1-rc.7.tgz. */
export const ENGINE = join(SITE_ROOT, 'node_modules/@zodiacs/engine');
/** vendor/zodiacs-engine-0.1.1-rc.6.tgz, extracted by run-all.sh: the positive control. */
export const RC6 = join(WORK, 'tgz-rc6/package');
/** The committed corpora of the engine brief's rules (grids A and L, the ERFA arbiter). */
export const CORPORA = join(SITE_ROOT, 'docs/platform/evidence/engine-beyond-swiss/corpora');

/** $WORK/<name>/, created if missing, with the trailing slash the tools concatenate onto. */
export function outDir(name) {
  const directory = join(WORK, name);
  mkdirSync(directory, { recursive: true });
  return `${directory}/`;
}

File: docs/platform/evidence/phase1-verdicts-2026-09-25/tools/lib/paths.py
"""Where the Phase 1 verdict tools read and write (the Python side of lib/paths.mjs).

Each path comes from the environment and falls back to a default:

  SITE_ROOT   the site repository: the root this folder sits in (tools/../../../../..)
  WORK        scratch output, outside the repository: <tempdir>/phase1-verdicts-2026-09-25
  SWISS_EPHE  the directory holding sepl_18.se1 and semo_18.se1: $WORK/ephe
  JPL_KERNEL  de440s.bsp, read only by s18/arbiter_moon.py: $WORK/de440s.bsp

Swiss Ephemeris output is written only under WORK; importing this module fails if WORK is the
repository or lies inside it.
"""
import os
import tempfile

TOOLS = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
SITE_ROOT = os.path.abspath(os.environ.get('SITE_ROOT') or os.path.join(TOOLS, '..', '..', '..', '..', '..'))
WORK = os.path.abspath(os.environ.get('WORK') or os.path.join(tempfile.gettempdir(), 'phase1-verdicts-2026-09-25'))
SWISS_EPHE = os.path.abspath(os.environ.get('SWISS_EPHE') or os.path.join(WORK, 'ephe'))
JPL_KERNEL = os.path.abspath(os.environ.get('JPL_KERNEL') or os.path.join(WORK, 'de440s.bsp'))
CORPORA = os.path.join(SITE_ROOT, 'docs', 'platform', 'evidence', 'engine-beyond-swiss', 'corpora')

_from_root = os.path.relpath(WORK, SITE_ROOT)
if not (_from_root == '..' or _from_root.startswith('..' + os.sep)):
    raise SystemExit(f'WORK ({WORK}) is inside the repository ({SITE_ROOT}); Swiss output must stay outside it')


def out_dir(name):
    """$WORK/<name>/, created if missing, with the trailing slash the tools concatenate onto."""
    directory = os.path.join(WORK, name)
    os.makedirs(directory, exist_ok=True)
    return directory + '/'
